src/System.IO.FileSystem/src/System/IO/FileInfo.Unix.cs

```diff
@@ -0,0 +1,16 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.IO
+{
+    partial class FileInfo
+    {
+        internal static unsafe FileInfo Create(string fullPath, string fileName, FileStatus fileStatus)
```

Comment:
`ref FileStatus fileStatus` ? (Do we have a rule of thumb for when a struct is large enough to be better by ref?)

Reply:
I considered it, but it isn't big and I wanted to err on the side of caution.

Reply:
@stephentoub / @jkotas is there a rule of thumb?

Reply:
> Do we have a rule of thumb for when a struct is large enough to be better by ref?

When it does not fit into a few registers (e.g. it is bigger than 16 bytes).

> but it isn't big

It is 100+ bytes if I am reading the code correctly. That qualifies as big struct to me. It would be definitely worth it to pass it byref here.